### PR TITLE
 [ENG-5985] Extract shared EXtrenal[AddonType]Service code 

### PR DESCRIPTION
**Pull request by opaduchak**
Extracted non storage-specific logic from ExternalStorageService to ExternalService
dependent on #82

### CODE REVIEW

File: addon_service/common/network.py [hunk 1]
```diff
@@ -60,7 +61,7 @@ def __init__(
         *,
         client_session: aiohttp.ClientSession,
         prefix_url: str,
-        account: db.AuthorizedStorageAccount,
+        account: "db.AuthorizedStorageAccount",
```
Comment (adlius) [inline addon_service/common/network.py:64]: Why stringified?

File: addon_service/common/network.py [hunk 2]
```diff
@@ -144,7 +145,7 @@ class _PrivateNetworkInfo(_PrivateInfo):
 
     # keep network constraints away from imps
     prefix_url: str
-    account: db.AuthorizedStorageAccount
+    account: "db.AuthorizedStorageAccount"
```
Comment (adlius) [inline addon_service/common/network.py:148]: Same here.

File: addon_service/external_storage_service/models.py [hunk 3]
```diff
@@ -1,106 +1,23 @@
...
 
 
-class ExternalStorageService(AddonsServiceBaseModel):
-    display_name = models.CharField(null=False)
+class ExternalStorageService(ExternalService):
     int_addon_imp = models.IntegerField(
         null=False,
         validators=[validate_storage_imp_number],
         verbose_name="Addon implementation",
     )
```
Comment (adlius) [inline addon_service/external_storage_service/models.py:12]: Can this be moved to the `ExternalService` class as well? I assume each external service, regardless of type, should have a addon implementation and therefore a int representing that implementation.

File: addon_service/abstract/external_storage/models.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        validators=[validate_service_type],
+        verbose_name="Service type",
+    )
+    supported_scopes = ArrayField(models.CharField(), null=True, blank=True)
+    api_base_url = models.URLField(blank=True, default="")
+
+    oauth1_client_config = models.ForeignKey(
+        "addon_service.OAuth1ClientConfig",
+        on_delete=models.SET_NULL,
+        related_name="external_storage_services",
```
Comment (adlius) [inline addon_service/abstract/external_storage/models.py:35]: `related_name` seems wrong.

File: addon_service/abstract/external_storage/models.py [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+        null=True,
+        blank=True,
+    )
+
+    oauth2_client_config = models.ForeignKey(
+        "addon_service.OAuth2ClientConfig",
+        on_delete=models.SET_NULL,
+        related_name="external_storage_services",
```
Comment (adlius) [inline addon_service/abstract/external_storage/models.py:43]: same here

**Review by adlius — CHANGES_REQUESTED** [review]
Some small nitpicks.

File: addon_service/external_storage_service/models.py (hunk 3 above)
Comment (opaduchak) [reply addon_service/external_storage_service/models.py:12]: This field has a validator which is checking whether this is **Storage** implementation, so it cannot be common for all external services.

**Review by adlius — APPROVED** [review]
Looks good.

**Pull request merged**